ipaclient/install/ipa_epn.py

```diff
@@ -0,0 +1,748 @@
+#
+# Copyright (C) 2020  FreeIPA Contributors see COPYING for license
+#
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""This tool prepares then sends email notifications to users
+   whose passwords are expiring in the near future.
+"""
+
+
+from __future__ import absolute_import, print_function
+
+import ast
+import grp
+import json
+import os
+import pwd
+import logging
+import smtplib
+
+from collections import deque
+from datetime import datetime, timedelta
+from email.utils import formataddr, formatdate
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from email.header import Header
+
+from ipaclient.install.client import is_ipa_client_installed
+from ipaplatform.paths import paths
+from ipalib import api, errors
+from ipalib.install import sysrestore
+from ipapython import admintool, ipaldap
+from ipapython.dn import DN
+
+from jinja2 import Environment, FileSystemLoader
+
+
+EPN_CONF = "/etc/ipa/epn.conf"
```

Comment:
I'm worried that ``/etc/ipa/epn.conf`` could be confused with a config file for IPA context. How about ``/etc/ipa/epn/epn.conf``?

Reply:
Typically context files are put into /etc/ipa. This is true with installer and server at least. The other contexts tend not to have config files.

Reply:
I thought that this was a special config file like ``custodia.conf``. Now I realized that it's actually a standard IPA config file.